Add menu employee loans advance paiement

lomamech opened the pull request:
- Adding interface for paiement Loans Advance 
- Add menu for paiement of loans, advances, installments

closes #6247

lomamech committed:
feature(Complex Voucher)
- Adding interface for paiement Loans Advance ...

closes #6247

lomamech committed:
feature(Complex Voucher)
- Add menu for paiement of loans, advances, installments

jniles commented on client/src/modules/vouchers/complex-voucher.html:
```diff
@@ -56,6 +56,12 @@
             </li>
 
+            <li role="menuitem">
+              <a href data-action="payment-employee" ng-click="ComplexVoucherCtrl.openAdvancesLoansInstallments()">
```
The tag `data-action="payment-employee"` is used above.   Can you change it to `employee-advance` or something?

jniles commented on client/src/modules/vouchers/templates/amount_paiment.grid.tmpl.html:
```diff
@@ -0,0 +1,8 @@
+    ng-model="row.entity.value"
+    ng-change="grid.appScope.onChanges()"
+    ng-disabled="row.entity.locked"
+    type="number"
```
Should we make this number `ng-min=0`?

jniles commented on client/src/modules/vouchers/toolkit/advances_loans_installments/advances_loans_installments.js:
```diff
@@ -0,0 +1,167 @@
+  };
+
+  function muteDisabledCells(grid, row) {
+    return (row.entity.locked) ? `text-muted strike` : '';
```
Do we need to even show these?  Or can we just filter them out?

I think a locked employee should never be paid or given an advance.

lomamech replied:
I thought it displayed like this to reassure the user, that there are employees who have a blocked status to prevent him from thinking that he does not exist

mbayopanda reviewed (commented):
Could you fix the currency when changing cash-boxes 

![image](https://user-images.githubusercontent.com/5445251/152327204-3da455bc-cbdf-4e11-b22d-466214de0151.png)

jniles reviewed (commented):
Looks pretty good to me.  I haven't tested this yet - I'll wait for live data to be available for a real simulation.

jniles commented on client/src/modules/vouchers/toolkit/advances_loans_installments/advances_loans_installments.modal.html:
```diff
@@ -0,0 +1,51 @@
+      cashbox-id = "ToolCtrl.searchQueries.cashbox_id"
+      name = "cashbox_id"
+      on-select-callback = "ToolCtrl.onSelectCashbox(cashbox)"
+      restrict-to-user = "false">
```
![image](https://user-images.githubusercontent.com/896472/152366902-8ff18228-26f8-4ed2-a041-c3d2f70d249d.png)

This should have `required` set on it, otherwise, it says "optional" at the top.

jniles replied:
Also, is there any way to make sure we never use the `auxiliary` cashbox?

jniles commented on client/src/modules/vouchers/toolkit/advances_loans_installments/advances_loans_installments.modal.html:
```diff
@@ -0,0 +1,51 @@
+      label="VOUCHERS.GLOBAL.DESTINATION_ACCOUNT"
+      on-select-callback="ToolCtrl.onSelectAccountCallback(account)"
+      exclude-title-accounts="true">
+      (<label translate>FORM.LABELS.DEBIT</label>)
```
Can we filter by `account_type_id = 2`?   Should this always be a `liability`?

lomamech replied:
I have a concern in Vanga the account he uses for social loans uses this account 
**510000 : Personal social loans** do you think i should add accounts of ASSET types

jniles replied:
Hm... does that make sense?  Or is Vanga doing poor accounting?

jniles replied:
I see you changed it to both asset and liability - you are right, it can be both.  I was wrong.

jniles commented on client/src/modules/vouchers/toolkit/advances_loans_installments/advances_loans_installments.js:
```diff
@@ -0,0 +1,167 @@
+    headerCellFilter : 'translate',
+    cellClass : muteDisabledCells,
+  }, {
+    field : 'value',
```
You should turn sorting off on both these columns - it could have problems down the line.

jniles reviewed (commented):
Some more comments now that I have tested this.

lomamech committed:
Bug: changing currency when cashbox change

lomamech committed:
improvement(component cashbox select)
- add filter for primary cashbox

lomamech committed:
improvement(Paiement advance loans)
- Select account of type is liability and asset
- Set enable sorting false in grid

jniles reviewed (approved):
LGTM.

bors r+
